engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/query/NadelQueryTransformer.kt

```diff
@@ -4,24 +4,51 @@ import graphql.nadel.Service
 import graphql.nadel.enginekt.NadelExecutionContext
 import graphql.nadel.enginekt.blueprint.NadelOverallExecutionBlueprint
 import graphql.nadel.enginekt.plan.NadelExecutionPlan
+import graphql.nadel.enginekt.transform.NadelTransform
 import graphql.nadel.enginekt.transform.NadelTransformFieldResult
 import graphql.nadel.enginekt.util.toBuilder
 import graphql.normalized.ExecutableNormalizedField
 
-class NadelQueryTransformer internal constructor(
+class NadelQueryTransformer private constructor(
```

Comment:
Ok so a lot of the change in this class is a refactor. The big issue with `NadelQueryTransformer` is that it used to be that it was PER Nadel instance. Now created it's per query.

This means that for a query, we store the `Service` and `ExecutionContext` etc. associated with that query as global vars. This greatly cleans things up and means instead of passing a `Continuation` object to the transforms, we can just expose this class as API.

e.g.

```kotlin
private suspend fun transformField(
    executionContext: NadelExecutionContext,
    transformContext: TransformContext,
    executionPlan: NadelExecutionPlan,
    service: Service,
    field: ExecutableNormalizedField,
): List<ExecutableNormalizedField>
```

is now

```kotlin
suspend fun transform(
    fields: List<ExecutableNormalizedField>,
): List<ExecutableNormalizedField>
```

because all those params are now stored as global props on the class.